fix-version.patch

```diff
@@ -1,9 +1,11 @@
+diff --git a/maint/version.m4 b/maint/version.m4
+index f72d55c..2464c4b 100644
 --- a/maint/version.m4
 +++ b/maint/version.m4
 @@ -36,6 +36,6 @@ m4_define([MPICH_RELEASE_DATE_m4],[unreleased development copy])dnl
  # standard, and does not include MPIX_ functions and C++ bindings.
  
- # last version: 4.0a1 - 13:11:1
+ # last version: 3.4.2 - 13:10:1
```

Comment:
Is this really rolling back to 3.4.2 and DSO 13.10.1?

Reply:
im not sure what is that. this is where they made this change in this comment:
https://github.com/pmodels/mpich/commit/98717ec2199c385e5e5b30d61fdc384ecfb0346c
https://github.com/pmodels/mpich/blob/main/maint/version.m4

Reply:
Hrm.  Yeah.  Interesting "lineage" they have going there where they first add the comment:
```
# last version: 4.0a1 - 13:11:1
```
in 4.0a1: 
https://github.com/pmodels/mpich/commit/6963590cd84b2746a87c7d2fced27b9996469f10#diff-d1a0995338191d56316b4a836ff05cbdbd1f99f2b38520e8bd980d2ec3bc581f
and then change it to:
```
# last version: 3.4.2 - 13:10:1
```
in 4.0a2:
https://github.com/pmodels/mpich/commit/98717ec2199c385e5e5b30d61fdc384ecfb0346c#diff-d1a0995338191d56316b4a836ff05cbdbd1f99f2b38520e8bd980d2ec3bc581f

I guess maybe the latter is a correction of the former.  In any case, if it really is their intention that 4.0 will have `libmpi.so.13.10.1`, then this fix-version.patch change looks fine.  I'm noting that it looks like mpich-4.0~a1-2.g032b3aeb2.el8.x86_64 (which I am assuming is ours) still has `libmpi.so.12.1.10` in it.  Maybe I am misunderstanding the `[13:10:1]` specification and that is not the `libmpi.so` library version.  Yeah, looking at the resulting build, it seems I am misunderstanding that nomenclature.  Makes me wonder how/what they are using it (for) now.

Also, isn't this merge an update to 4.0a2 which needs `*.spec` file updates also?

Reply:
done.